modules/packages/MPI.chpl

```diff
@@ -260,6 +265,20 @@ module MPI {
      processes (world size)
      */
   proc initialize() {
+    // If we are running using the uGNI layer, then the following hack
+    // appears to be necessary in order to run MPI, as well as Chapel
+    // See : https://hpcrdm.lbl.gov/pipermail/upc-users/2014-May/002061.html
+    if isUGNI {
+      coforall loc in Locales do on loc {
+          // This must be done on all locales!
+          var pmiGniCookie = C_Env.getenv("PMI_GNI_COOKIE") : string;
+          if !pmiGniCookie.isEmptyString() {
+            C_Env.setenv("PMI_GNI_COOKIE",("%i".format(pmiGniCookie:int+1)).c_str(),1);
+          }
```

Comment:
That's interesting..... it might also explain issues I had trying to get the ``gasnet+aries`` case running when I tried it a few months ago. It would be great if this fixes both cases....